client/state/document-head/reducer.js

```diff
@@ -18,7 +18,7 @@ import {
 export function title( state = '', action ) {
 	switch ( action.type ) {
 		case DOCUMENT_HEAD_TITLE_SET:
-			return action.title;
+			return action.title || state;
```

Comment:
Ah, I guess this is why it's okay to dispatch with `false`? But as mentioned in a previous pull request, there are cases where we want to intentionally reset the title to `undefined` (#5897). In any case, I'd rather we avoid dispatching altogether if we don't intend for the title to be used.